mava/systems/jax/ippo/networks.py

```diff
@@ -231,37 +256,41 @@ def critic_fn(inputs: jnp.ndarray) -> networks_lib.FeedForwardNetwork:
         )
         return critic_network(inputs)
 
-    # Transform into pure functions.
-    policy_fn = hk.without_apply_rng(hk.transform(policy_fn))
-    critic_fn = hk.without_apply_rng(hk.transform(critic_fn))
-
     dummy_obs = utils.zeros_like(environment_spec.observations.observation)
     dummy_obs = utils.add_batch_dim(dummy_obs)  # Dummy 'sequence' dim.
 
     network_key, key = jax.random.split(key)
-    policy_params = policy_fn.init(network_key, dummy_obs)  # type: ignore
+
+    if len(policy_recurrent_layer_sizes) > 0:
+        policy_state = initial_state_fn.apply(None)
+
+        policy_params = policy_fn.init(network_key, [dummy_obs, policy_state])  # type: ignore
+    else:
+        policy_state = None
+        policy_params = policy_fn.init(network_key, dummy_obs)  # type: ignore
 
     network_key, key = jax.random.split(key)
     critic_params = critic_fn.init(network_key, dummy_obs)  # type: ignore
 
     # Create PPONetworks to add functionality required by the agent.
-    return make_ppo_networks(
+    return PPONetworks(
```

Comment:
Thanks for the refactor on this class! It was very scattered before. I guess this is more a comment for the future but as we add more algs, we would not return PPONetworks here and I think we need to be clever with how we use inheritance and create a base class for PPONetworks

Reply:
Yes, I agree. This networks file is currently only for IPPO. But if we want a base networks class we need to find a format that is more general, yes.